public/js/components/SortingBar.jsx

```diff
@@ -9,18 +9,36 @@ export default class SortingBar extends React.Component {
     ArticleActions.fetchArticles(sort)
   }
 
+  constructor() {
+    super()
+    this.state = {
+      newActive: null,
+      topActive: null,
+    }
+  }
+
+  setNewActive() {
+    this.setState({ newActive: true, topActive: null })
+  }
+
+  setTopActive() {
+    this.setState({ newActive: null, topActive: true })
+  }
+
   render() {
     return (
       <ButtonGroup>
         <Button
-          onClick={() => SortingBar.changeSort('NEW')}
+          onClick={() => { SortingBar.changeSort('NEW'); this.setNewActive() }}
           bsStyle="info"
           bsSize="small"
+          active={this.state.newActive}
         >Newest</Button>
         <Button
-          onClick={() => SortingBar.changeSort('TOP')}
+          onClick={() => { SortingBar.changeSort('TOP'); this.setTopActive() }}
```

Comment:
Also in the constructor you should do bindings to ensure that `this` in the `setTopActive` is always referring to `SortingBar`